website/docs/r/cluster.html.markdown

```diff
@@ -206,6 +206,11 @@ Note free tier (M0) creation is not supported by the Atlas API and hence not sup
 * `auto_scaling_disk_gb_enabled` - (Optional) Specifies whether disk auto-scaling is enabled. The default is true.
     - Set to `true` to enable disk auto-scaling.
     - Set to `false` to disable disk auto-scaling.
+* `auto_scaling_compute_enabled` - (Optional) Specifies whether cluster tier auto-scaling is enabled. The default is false.
+    - Set to `true` to enable cluster tier auto-scaling. If enabled, you must specify a value for `providerSettings.autoScaling.compute.maxInstanceSize`.
+    - Set to `false` to disable cluster tier auto-scaling.
+* `auto_scaling_compute_scale_down_enabled` - (Optional) Set to `true` to enable the cluster tier to scale down. This option is only available if `autoScaling.compute.enabled` is `true`.
+    - If this option is enabled, you must specify a value for `providerSettings.autoScaling.compute.minInstanceSize`
```

Comment:
We need an important note in the doc that if auto_scaling_compute_enabled then Atlas will automatically scale up to the maximum provided and down to the minimum, if provided.  This will cause the value of instanceSizeName returned to potential be different than what is specified in the Terraform config and if one then applies a plan, not noting this, Terraform will scale the cluster back down to the original instanceSizeName value.  To prevent this a lifecycle customization should be used, i.e.:
lifecycle {
  ignore_changes = instanceSizeName
}

We should also test that a lifecycle customization works as expected, thought that will be tough to auto trigger a scaling event so one could test manually by running a terraform config, change the instance size in the UI to between the autoscale values and see if the lifecycle customerization works properly.

Reply:
I pushed with note important about lifecycle and added a similar situation,its test, thank you so much, melissa !